eth/backend.go

```diff
@@ -83,63 +75,43 @@ func (b *backend) PendingNonceAt(ctx context.Context, account common.Address) (u
 }
 
 func (b *backend) SendTransaction(ctx context.Context, tx *types.Transaction) error {
-	sendErr := b.Client.SendTransaction(ctx, tx)
+	// Use the transaction manager instead of the ethereum client
+	if err := b.tm.SendTransaction(ctx, tx); err != nil {
+		return err
+	}
 
 	msg, err := tx.AsMessage(b.signer)
 	if err != nil {
 		return err
 	}
 	sender := msg.From()
 
-	txLog, err := b.newTxLog(tx)
-	if err != nil {
-		txLog.method = "unknown"
-	}
-
-	if sendErr != nil {
-		glog.Infof("\n%vEth Transaction%v\n\nInvoking transaction: \"%v\". Inputs: \"%v\"   \nTransaction Failed: %v\n\n%v\n", strings.Repeat("*", 30), strings.Repeat("*", 30), txLog.method, txLog.inputs, sendErr, strings.Repeat("*", 75))
-		return sendErr
-	}
-
 	// update local nonce
 	b.nonceManager.Lock(sender)
 	b.nonceManager.Update(sender, tx.Nonce())
 	b.nonceManager.Unlock(sender)
 
-	glog.Infof("\n%vEth Transaction%v\n\nInvoking transaction: \"%v\". Inputs: \"%v\"  Hash: \"%v\". \n\n%v\n", strings.Repeat("*", 30), strings.Repeat("*", 30), txLog.method, txLog.inputs, tx.Hash().String(), strings.Repeat("*", 75))
-
 	return nil
 }
 
 func (b *backend) SuggestGasPrice(ctx context.Context) (*big.Int, error) {
 
 	// Use the gas price monitor instead of the ethereum client to suggest a gas price
 	gp := b.gpm.GasPrice()
+	maxGp := b.gpm.MaxGasPrice()
 
-	if b.maxGasPrice != nil && gp.Cmp(b.maxGasPrice) > 0 {
+	if maxGp != nil && gp.Cmp(maxGp) > 0 {
 		return nil, fmt.Errorf("current gas price exceeds maximum gas price max=%v GWei current=%v GWei",
-			FromWei(b.maxGasPrice, params.GWei),
+			FromWei(maxGp, params.GWei),
 			FromWei(gp, params.GWei),
 		)
 	}
 
 	return gp, nil
 }
 
-func (b *backend) SetMaxGasPrice(gp *big.Int) {
-	b.Lock()
-	defer b.Unlock()
-	b.maxGasPrice = gp
-
-	if gp != nil && monitor.Enabled {
-		monitor.MaxGasPrice(gp)
-	}
-}
-
-func (b *backend) MaxGasPrice() *big.Int {
-	b.RLock()
-	defer b.RUnlock()
-	return b.maxGasPrice
+func (b *backend) GasPriceMonitor() *GasPriceMonitor {
```

Comment:
Since we're removing the max gas price setters/getters from backend should we also remove the min gas price setters/getters with the expectation that the appropriate setters/getters can be called on the GasPriceMonitor returned here?

Reply:
Oh yeah forgot to remove these, the calls were already updated to call them on the GPM instead

Reply:
fixed in 91ceac9